src/key.js

```diff
@@ -297,25 +253,37 @@ Key.prototype.armor = function() {
   return armor.encode(type, this.toPacketlist().write());
 };
 
+function isValidSigningKeyPacket(keyPacket, signature, date=new Date()) {
+  const normDate = util.normalizeDate(date);
+  return keyPacket.algorithm !== enums.read(enums.publicKey, enums.publicKey.rsa_encrypt) &&
+         keyPacket.algorithm !== enums.read(enums.publicKey, enums.publicKey.elgamal) &&
+         keyPacket.algorithm !== enums.read(enums.publicKey, enums.publicKey.ecdh) &&
+         (!signature.keyFlags ||
+          (signature.keyFlags[0] & enums.keyFlags.sign_data) !== 0) &&
+         signature.verified && !signature.revoked && !signature.isExpired(normDate) &&
+         (normDate === null || (keyPacket.created <= normDate && normDate < getExpirationTime(keyPacket, signature, Infinity)));
+}
+
 /**
- * Returns first key packet or key packet by given keyId that is available for signing or signature verification
+ * Returns first key packet or key packet by given keyId that is available for signing and verification
+ *
+ * NOTE: call verifyKeyPackets before calling this function.
```

Comment:
I don't think that's a good idea, for the same reason that we don't validate the parameters pretty much anywhere in this file: these aren't high level API functions.

Reply:
I brought it up because I remember there was some confusion related to functions depending on verifyPrimaryUser, and we wanted to throw errors if these dependent functions weren't previously called for debugging purposes.

Reply:
Yeah I agree with Sunny. Silent errors because some function wasn't previously called are bad.